packages/data-stores/src/plans/hooks/use-pricing-meta-for-grid-plans.ts

```diff
@@ -175,7 +175,52 @@ const usePricingMetaForGridPlans = ( {
 				}
 
 				/**
-				 * 2. Original and Discounted prices for plan available for purchase.
+				 * 2. Original and Discounted prices for site-specific plans available for purchase
+				 */
+				if ( siteId && planAvailabilityForPurchase[ planSlug ] ) {
+					const originalPrice = {
+						monthly: getTotalPrice(
+							sitePlan?.pricing.originalPrice.monthly,
+							storageAddOnPriceMonthly
+						),
+						full: getTotalPrice( sitePlan?.pricing.originalPrice.full, storageAddOnPriceYearly ),
+					};
+
+					// Do not return discounted prices if discount is due to plan proration
+					if ( sitePlan?.pricing.discountReasonCode === 'recent-plan-proration' ) {
```

Comment:
I think this is the part that I wanted us to parameterize. So to turn this condition into:

```
if ( ! withProration && sitePlan?.pricing.discountReasonCode === 'recent-plan-proration' ) {
```

Reply:
Ah I see you are already on this from your other comment :-)